pkgs/by-name/cu/cutadapt/package.nix

```
@@ -0,0 +1,66 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+
+  # update
+  nix-update-script,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "cutadapt";
+  version = "5.2";
+  pyproject = true;
+  build-system = with python3Packages; [
+    setuptools
+    setuptools-scm
+    cython
+  ];
+
+  src = fetchFromGitHub {
+    owner = "marcelm";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-CA6wp0By+TivsghR0tWHPvqXdSSvcMCzFOOnrUydKUI=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    xopen
+    dnaio
+  ];
+
+  pythonImportsCheck = [ "cutadapt" ];
+
+  preCheck = ''
+    # some tests use the executable
+    export PATH="${placeholder "out"}/bin:$PATH"
```

shell already has access to $out
```suggestion
    export PATH="$out/bin:$PATH"
```